src/charts/Donut.js

```diff
@@ -0,0 +1,80 @@
+import Responsive from '../components/Responsive'
+import ChartSvg from '../components/ChartSvg'
+import { CATEGORICAL } from '../utils/colors'
+
+const TAU = Math.PI * 2
+
+const CircleSegment = ({ width, height, hole=.5, start=0, end=.3, ...props }) => {
+  const center = { x: width/2, y: height/2 }
+  const radius = Math.min(width, height) / 2
+  const innerRadius = radius * hole
+  const strokeWidth = radius - innerRadius
+
+  const innerA = {
+    x: center.x + Math.cos(start * TAU) * innerRadius,
+    y: center.y + Math.sin(start * TAU) * innerRadius,
+  }
+
+  const innerB = {
+    x: center.x + Math.cos(end * TAU) * innerRadius,
+    y: center.y + Math.sin(end * TAU) * innerRadius,
+  }
+
+  const outerA = {
+    x: center.x + Math.cos(start * TAU) * radius,
+    y: center.y + Math.sin(start * TAU) * radius,
+  }
+
+  const outerB = {
+    x: center.x + Math.cos(end * TAU) * radius,
+    y: center.y + Math.sin(end * TAU) * radius,
+  }
+  
+  return  <path d={
+    `M${innerA.x} ${innerA.y}
+     L${outerA.x} ${outerA.y}
+     A${radius} ${radius} 1 ${Number(end - start >= .5)} 1 ${outerB.x} ${outerB.y}
+     L${innerB.x} ${innerB.y}
+     A${innerRadius} ${innerRadius} 0 ${Number(end - start >= .5)} 0 ${innerA.x} ${innerA.y}`
+  } {...props} />
+}
+
+const Donut = ({
+  width,
+  height = '100%',
+  data = [],
+  colors = CATEGORICAL,
+  segmentStyle,
+  segmentStyles,
+  offset=0,
+  hole=.5,
+  ...props
+}) => {
+  const sum = data.reduce((a, b) => a + b, 0)
+  const portions = data.map((x, i) => {
+    const previousSum = data.slice(0, i).reduce((a, b) => a + b, 0)
+    return { start: previousSum / sum, end: (previousSum + x) / sum }
+  })
+
+  return (
+    <Responsive style={{ width, height }}>
+      {({ width: autoWidth, height: autoHeight }) => (
+        <ChartSvg width={autoWidth} height={autoHeight} {...props} >
+          {portions.map((portion, i) =>
+            <CircleSegment
+              key={i}
+              fill={colors?.[i % colors?.length]}
+              width={autoWidth}
+              height={autoHeight}
+              start={portion.start + offset + (data.length === 1 ? 0.000001 : 0)}
```

Comment:
We should probably figure this out

Reply:
I did some reading and the path arc motion probably just won't cut it here.

> Complete circles and ellipses are the only shapes that SVG arcs have trouble drawing. Because the start and end points for any path going around a circle are the same point, there are an infinite number of circles that could be chosen, and the actual path is undefined. It's possible to approximate them by making the start and end points of the path slightly askew, and then connecting them with another path segment. For example, it's possible to make a circle with an arc for each semi-circle. At that point, it's often easier to use a real [<circle>](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/circle) or [<ellipse>](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/ellipse) node instead.

I think the only real solution is to either do as we have, to use a `<circle/>` or duplicate the data, colour and style of the data when there is only a single data point. In the case of `data.length === 1` then the proportion calculations are redundant. In that case we could render something like
```js
<>
  <CircleSegment start={0} end={0.5} fill={colours[0]} style={segmentStyles[0]} />
  <CircleSegment start={.5} end={1} fill={colours[0]} style={segmenStyles[0]} />
</>
```

Any thoughts?

Reply:
Ooh yeah I like this, if we can just display 2 segments and make them both use the first color, I think that's ideal